### PR TITLE
fixes #10030 -find templates in folders

### PR DESCRIPTION
**Pull request by shlomizadok**
Pending on merging of fog/fog#3531
(Unless we want to override this on Foreman, with fog_extension

### CODE REVIEW

**Comment by lzap** [comment]
Ping me once it is ready to be tested.

**Comment by dLobatog** [comment]
Also pending a new release of fog with the patch, and we including it in Foreman :smile:

**Comment by domcleal** [comment]
As per my ticket comment, I'd suggest applying a workaround in the meantime to fog_extensions and then if the other PR is merged, when it's released and when it's in Foreman, apply the correct fix.

**Comment by shlomizadok** [comment]
Added an initializer with override to fog method. The initializer should be removed when fog v1.30.0 is out and used by Foreman.

File: config/initializers/fog_overrides.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'fog/vsphere/requests/compute/get_virtual_machine'
```
Comment (domcleal) [inline config/initializers/fog_overrides.rb]: Isn't this going to fail if Fog isn't installed?  Can we use existing fog extensions?

**Comment by shlomizadok** [comment]
I couldn't override the method with fog extensions (seems that fog extensions adds extra methods to fog, but doesn't override them [am I correct @ohadlevy ?])
Added an if defined? to ensure fog is present

**Comment by domcleal** [comment]
Fog extensions are just a .include, so you should at least be able to alias/chain methods.  I thought it might overwrite as well by default though?

File: config/initializers/fog_overrides.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+if defined?(Fog)
+  require 'fog/vsphere/requests/compute/get_virtual_machine'
+  Fog::Compute::Vsphere::Real.class_eval do
+    protected
```
Comment (domcleal) [inline config/initializers/fog_overrides.rb]: rubocop warns "Keep a blank line before and after `protected`."

**Comment by shlomizadok** [comment]
Agreed. Moved to fog_extensions.

**Comment by dLobatog** [comment]
@lzap It should be ready to be tested, mind to give it a spin?

**Comment by discreet** [comment]
This fixed my issue deploying via VMware templates in a folder.

**Comment by domcleal** [comment]
Merged as 0af12f7, thanks @shlomizadok and @discreet!

**Comment by lzap** [comment]
Sorry I did not managed to get to a VmWare with nested folders.

**Pull request closed without merging**